src/Commands/Commands.php

```diff
@@ -46,4 +57,52 @@ public function fetchTrackingNumber($tracking_pin) {
     $this->output->writeln(var_export($tracking_summary, TRUE));
   }
 
+  /**
+   * Get rates for the provided postal codes and package weight.
+   *
+   * @param array $options
+   *   An array of options.
+   *
+   * @command commerce-canadapost-get-rates
+   *
+   * @option origin_postal_code
+   *  The origin postal code.
+   *  Defaults to H2B1A0.
+   * @option postal_code
+   *  The destination postal code.
+   *  Defaults to K1K4T3.
+   * @option weight
+   *  The weight (in grams) of the shipment.
+   *  Defaults to 1.
+   *
+   * @usage commerce-canadapost-get-rates
+   *   Get the shipping rates for a package (1g) from H2B1A0 to K1K4T3.
+   * @usage commerce-canadapost-get-rates
+   *   Alias to get the shipping rates for a package (1g) from H2B1A0 to
+   * K1K4T3.
+   * @usage commerce-cp-gr H0H0H0
+   *   Get the shipping rates for a package (1g) from H0H0H0 to K1K4T3.
+   * @usage commerce-cp-gr H0H0H0 K1V1J8
+   *   Get the shipping rates for a package (1g) from H0H0H0 to K1V1J8.
+   * @usage commerce-cp-gr H0H0H0 K1V1J8 100
+   *   Get the shipping rates for a package (100g) from H0H0H0 to K1V1J8.
+   *
+   * @aliases commerce-cp-gr
+   */
+  public function getRates(
+    array $options
+      = [
+        'origin_postal_code' => 'H2B1A0',
+        'postal_ode' => 'K1K4T3',
+        'weight' => 1,
+      ]
+  ) {
```

Comment:
Typo: should be `postal_code`.

Passing on the default options this way is problematic:

1. Poor readability, it's not great practice to do that in the function arguments, even though it is possible.
2. You cannot provide only one option without replacing all of them e.g. if you call `getRates(['postal_code' => 'ANOTHER']);` the options passed will be just that, the rest of the options won't be available in the array.

Those issues are solved by doing:

```
public function getRates(array $options) {
  $options = array_merge(
    [
      'origin_postal_code' => ...,
      'postal_code' => ...,
      'weight' => 1,
    ],
    $options
  );
}

Also, I haven't tested this but options are called explicitly e.g. `--postal-code="CODE"` otherwise they are arguments. Have you tested that this works as described in the usage annotations?

Reply:
so by changing `RateService::getRates()` to require a $shipment, whats the best way to create a `Shipment` without actually saving an order to the database:

eg:
```php
    // temp code get store and user from storage service
    $store = Store::load(1);
    $user = User::load(1);
    $order = Order::create([
      'type' => 'default',
      'state' => 'draft',
      'mail' => $user->getEmail(),
      'uid' => $user->id(),
      'store_id' => $store->id(),
    ]);

    $order->save();
    // couldnt find where to add the shipping info, still looking into it
    $shipment = Shipment::create([
      'type' => 'default',
      'items' => [
        new ShipmentItem([
          'order_item_id' => 10,
          'title' => 'T-shirt (red, large)',
          'quantity' => 1,
          'weight' => new Weight('10', 'kg'),
          'declared_value' => new Price('15.00', 'USD'),
        ]),
      ],
      'order_id' => $order->id(),
    ]);

    $shipment->save();

    $rates = $this->ratingApi->getRates($shipment);
```

Reply:
You can have 2 methods on the service:
- `getRates` as it used to be.
- `getRatesForShipment` which would fetching the details from the shipment and then calling the `getRates` function.

But I'd suggest removing this Drush command for now so that we can make a release (since it also has an issue its arguments). We can improve on this later.

Reply:
Also, you should better completely forget the `Entity::` syntax e.g. `Shipment::create` should be `$storage->create`.